substrate/client/transaction-pool/src/fork_aware_txpool/view_store.rs

```diff
@@ -338,6 +338,40 @@ where
 		self.active_views.read().is_empty() && self.inactive_views.read().is_empty()
 	}
 
+	/// Searches in the view store for the first descendant view by iterating through the fork of
+	/// the `at` block, up to the provided `block_number`.
+	///
+	/// Returns with a maybe pair of a view and a set of enacted blocks when the first view is
+	/// found.
+	pub(super) fn find_view_descendent_upto_number(
+		&self,
+		at: &HashAndNumber<Block>,
+		block_number: <<Block as BlockT>::Header as Header>::Number,
+	) -> Option<(Arc<View<ChainApi>>, Vec<Block::Hash>)> {
+		let mut enacted_blocks = Vec::new();
+		let mut at_hash = at.hash;
+		let mut at_number = at.number;
+
+		// Search for a view that can be used to get and return an approximate ready
+		// transaction set.
+		while at_number > block_number {
+			// Found a view, stop searching..
+			if let Some((view, _)) = self.get_view_at(at_hash, true) {
+				return Some((view, enacted_blocks));
+			}
+
+			enacted_blocks.push(at_hash);
+
+			// Move up into the fork. Return with no view and an empty enacted blocks
+			// list if we can't access the header of the current block.
+			let header = self.api.block_header(at_hash).ok().flatten()?;
+			at_hash = *header.parent_hash();
+			at_number = at_number.saturating_sub(1u32.into());
```

Comment:
Would that work?
```suggestion
			at_number = at_number.saturating_sub(One::one());
```

Reply:
it works: [2570fa3](https://github.com/paritytech/polkadot-sdk/pull/8533/commits/2570fa35c3c0e202b6e3cd2719ba4ce22cb3b44d).